src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1113,6 +1116,108 @@ class FastRandomStringGenerator : public StringGenerator {
     }
 };
 
+// The following is a replacement for partial template specialization for functions, which is not
+// currently supported in C++.
+template <typename F, typename T>
+struct convert{};
+
+// Conversion used between int, long, double
+template <typename F, typename T>
+T convertToNumeric(convert<F, T>, const F& from)
+{
```

Comment:
Nit: formatting - please run `clang-format` (the versions aren't standard so if you get a bunch of diffs, just do your best to match the existing style; the brace on the new line threw this off for me.